bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliDt8DeviceHandler.java

```diff
@@ -0,0 +1,187 @@
+                }
+
+                DaliAddress readAddress = address;
+                if (readDeviceTargetId != null) {
+                    readAddress = DaliAddress.createShortAddress(readDeviceTargetId);
+                }
+                // Write argument 0xE9 (query red dimlevel) to DTR0 and set DT8
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDTR0Command(0xe9));
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDeviceTypeCommand(8));
+                // Red component is returned as result
+                CompletableFuture<@Nullable NumericMask> responseRed = getBridgeHandler().sendCommandWithResponse(
+                        DaliStandardCommand.createQueryColorValueCommand(readAddress), DaliResponse.NumericMask.class);
+                // Write argument 0xEA (query green dimlevel) to DTR0 and set DT8
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDTR0Command(0xea));
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDeviceTypeCommand(8));
+                // Green component is returned as result
+                CompletableFuture<@Nullable NumericMask> responseGreen = getBridgeHandler().sendCommandWithResponse(
+                        DaliStandardCommand.createQueryColorValueCommand(readAddress), DaliResponse.NumericMask.class);
+                // Write argument 0xEB (query blue dimlevel) to DTR0 and set DT8
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDTR0Command(0xeb));
+                getBridgeHandler().sendCommand(DaliStandardCommand.createSetDeviceTypeCommand(8));
+                // Blue component is returned as result
+                CompletableFuture<@Nullable NumericMask> responseBlue = getBridgeHandler().sendCommandWithResponse(
+                        DaliStandardCommand.createQueryColorValueCommand(readAddress), DaliResponse.NumericMask.class);
+
+                CompletableFuture.allOf(responseRed, responseGreen, responseBlue).thenAccept(x -> {
+                    @Nullable
+                    NumericMask r = responseRed.join(), g = responseGreen.join(), b = responseBlue.join();
+                    if (r != null && !r.mask && g != null && !g.mask && b != null && !b.mask) {
+                        final int rValue = r.value != null ? r.value : 0;
+                        final int gValue = g.value != null ? g.value : 0;
+                        final int bValue = b.value != null ? b.value : 0;
+                        updateState(channelUID, HSBType.fromRGB(rValue, gValue, bValue));
+                    }
+                }).exceptionally(e -> {
+                    logger.warn("Error querying device status: {}", e.getMessage());
+                    return null;
+                });
+
+            } else {
+                super.handleCommand(channelUID, command);
+            }
+        } catch (DaliException e) {
+            logger.error("Dali Exception", e);
```

Comment:
Will this make the binding stop working? Otherwise log level should be reduced. Perhaps it can even be removed as the next line will also be logged by the framework. For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Removed.